exercises/bank-account/test/Tests.hs

```diff
@@ -24,50 +29,40 @@ updateBalance account amount
 The initial balance of the bank account should be 0.
 -}
 
-exitProperly :: IO Counts -> IO ()
-exitProperly m = do
-  counts <- m
-  exitWith $ if failures counts /= 0 || errors counts /= 0 then ExitFailure 1 else ExitSuccess
+main :: IO ()
+main = hspecWith defaultConfig {configFastFail = True} specs
+
+specs :: Spec
+specs = describe "bank-account" $ do
 
-testCase :: String -> Assertion -> Test
-testCase label assertion = TestLabel label (TestCase assertion)
+    -- As of 2016-08-01, there was no reference file
+    -- for the test cases in `exercism/x-common`.
 
-main :: IO ()
-main = exitProperly $ runTestTT $ TestList
-  [ withBank "initial balance is 0" $
-    checkReturn (Just 0) . getBalance
-  , withBank "incrementing and checking balance" $ \acct -> do
-    checkReturn (Just 0) $ getBalance acct
-    checkReturn (Just 10) $ incrementBalance acct 10
-    checkReturn (Just 10) $ getBalance acct
-  , withBank "incrementing balance from other processes then checking it\
-             \from test process" $ \acct -> do
-    replicateM 20 (incrementProc acct) >>= mapM_ (void . takeMVar)
-    checkReturn (Just 20) (getBalance acct)
-  , testCase "closed banks hold no balance" $ do
-    acct <- openAccount
-    checkReturn (Just 0) (getBalance acct)
-    checkReturn (Just 10) $ incrementBalance acct 10
-    closeAccount acct
-    checkReturn Nothing (getBalance acct)
-    checkReturn Nothing $ incrementBalance acct 10
-  ]
+    it "initial balance is 0" $ do
+        account  <-  openAccount
+        getBalance   account `shouldReturn` Just 0
+        closeAccount account
 
-incrementProc :: BankAccount -> IO (MVar ())
-incrementProc acct = do
-  v <- newEmptyMVar
-  void . forkIO $ do
-    void (incrementBalance acct 1)
-    putMVar v ()
-  return v
+    it "incrementing and checking balance" $ do
+        account    <-    openAccount
+        getBalance       account    `shouldReturn` Just  0
+        incrementBalance account 10 `shouldReturn` Just 10
+        getBalance       account    `shouldReturn` Just 10
+        closeAccount     account
 
-checkReturn :: (Show a, Eq a) => a -> IO a -> IO ()
-checkReturn expect test = do
-  v <- test
-  expect @=? v
+    it "incrementing balance from other processes then checking it from test process" $ do
+        account  <-  openAccount
+        replicateM_ 20 $ do
+            v <- newEmptyMVar
+            _ <- forkIO $ incrementBalance account 1 >> putMVar v ()
+            takeMVar v
```

Comment:
I'm a bit unsure on this, but can it be the case that the previous code would potentially perform `incrementBalance` in parallel, but this code can never do that (since the `takeMVar` is in the same `replicateM_` as the `forkIO`?)